app/models/background.rb

```diff
@@ -19,42 +20,72 @@ def validate_size
   end
 
   def summarize
-    results = TaxonCount.connection.select_all("
+    rows = TaxonCount.connection.select_all("
       SELECT
         tax_id,
         count_type,
         tax_level,
         @adjusted_total_reads := (total_reads - IFNULL(total_ercc_reads, 0)) * IFNULL(fraction_subsampled, 1.0),
-        sum((1.0*1e6*count)/@adjusted_total_reads) AS sum_rpm,
-        sum((1.0*1e6*count*1e6*count)/(@adjusted_total_reads*@adjusted_total_reads)) AS sum_rpm2
+        (1.0*1e6*count)/@adjusted_total_reads as rpm,
+        (1.0*1e6*count*1e6*count)/(@adjusted_total_reads*@adjusted_total_reads) AS rpm2
       FROM `taxon_counts`
       INNER JOIN `pipeline_runs` ON
         `pipeline_runs`.`id` = `taxon_counts`.`pipeline_run_id`
       WHERE (pipeline_run_id in (select pipeline_run_id from backgrounds_pipeline_runs where background_id = #{id}))
-      GROUP BY tax_id, count_type, tax_level
-    ").to_hash
+      ORDER BY tax_id, count_type, tax_level
+    ").to_a
     n = pipeline_runs.count
     date = DateTime.now.in_time_zone
-    results.each do |h|
-      h[:background_id] = id
-      h[:created_at] = date
-      h[:updated_at] = date
-      h[:mean] = (h["sum_rpm"] || 0.0) / n.to_f
-      h[:stdev] = compute_stdev(h["sum_rpm"] || 0.0, h["sum_rpm2"] || 0.0, n)
+    key = ""
+    res = {}
```

Comment:
would prefer a name like `single_taxon_results` (and then `all_taxa_results` instead of `results`) -- it would do a lot for readability -- took me sometime to figure out what the code does

Reply:
i think taxon_results and result_list is good enough. I don't like long names.